notebooks_community/FuRBO/FuRBO.ipynb

```diff
@@ -0,0 +1,1011 @@
+    "        self.radius: float = 1.0                                                   # Percentage around which the trust region is built\n",
+    "        self.radius_min: float = 0.5**7                                            # Minimum percentage for trust region\n",
+    "\n",
+    "        # Trust region updating \n",
+    "        self.failure_counter: int = 0       # Counter of failure points to asses how algorithm is going\n",
+    "        self.success_counter: int = 0       # Counter of success points to asses how algorithm is going\n",
+    "        self.success_tolerance: int = 2     # Success tolerance for \n",
+    "        self.failure_tolerance: int = 3     # Failure tolerance for\n",
+    "        \n",
+    "        # Tensor to save current batch information\n",
+    "        self.batch_X: Tensor        # Current batch to evaluate: X values\n",
+    "        self.batch_Y: Tensor        # Current batch to evaluate: Y value\n",
+    "        self.batch_C: Tensor        # Current batch to evaluate: C values\n",
+    "            \n",
+    "        # Stopping criteria information\n",
+    "        self.n_iteration: int = n_iteration     # Maximum number of iterations allowed\n",
+    "        self.it_counter: int = 0                # Counter of iterations for stopping\n",
+    "        self.finish_trigger: bool = False       # Trigger to stop optimization\n",
+    "        self.failed_GP : bool = False           # Flag to pass to failed_GP in FuRBORestart\n",
+    "        \n",
+    "        # Restart criteria information\n",
+    "        self.restart_trigger: bool = False\n",
+    "        \n",
+    "        # Sobol sampler engine\n",
+    "        self.sobol = SobolEngine(dimension=self.dim, scramble=True, seed=1)\n",
+    "        \n",
+    "    # Update the status\n",
+    "    def update(self,\n",
+    "               X_next,          # Samples X (input values) to update the status\n",
+    "               Y_next,          # Samples Y (objective value) to update the status\n",
+    "               C_next,          # Samples C (constraints values) to update the status\n",
+    "               **tkwargs):\n",
+    "        \n",
+    "        '''Function to update optimization status'''\n",
+    "        \n",
+    "        # Merge current batch with previously evaluated samples\n",
+    "        if not hasattr(self, 'X'):\n",
+    "            # If there are no previous samples, declare the Tensors\n",
+    "            self.X = X_next\n",
+    "            self.Y = Y_next\n",
+    "            self.C = C_next\n",
+    "        else:\n",
+    "            # Else, concatenate the new batch to the previous samples\n",
+    "            self.X = torch.cat((self.X, X_next), dim=0)\n",
+    "            self.Y = torch.cat((self.Y, Y_next), dim=0)\n",
+    "            self.C = torch.cat((self.C, C_next), dim=0)\n",
+    "\n",
+    "        # update GPR surrogates\n",
```

Comment:
Why is there a try/catch here? Does the model fitting fail, and if so, do we have a sense on why that is?